packages/ui/src/common/utils.ts

```diff
@@ -9,19 +9,24 @@ export const isString = (something: unknown): something is string => typeof some
 export const isRecord = (something: unknown): something is Record<string, any> =>
   typeof something === 'object' && something !== null
 
+interface EqualsOption {
+  checkExtraKeys?: boolean
+  deep?: boolean
+}
+
 export const objectEquals = <T extends Record<string, any>>(
```

Comment:
This one would require a test. There's a lot going on and a test could clear up how it should be used.